### PR TITLE
Feature/rewrite api

### PR DESCRIPTION
**Pull request by ikrivosheev**
Hello, thank you for the great library! The PR improves and makes API more ergonomic and fix problem with performance.
Structure is: `Cabinet` -> `FolderEntries` -> `FolderReader` -> `FileEntries` -> `FileReader`.

## Changes
1. Replace `actions-rs` to `dtolnay/rust-toolchain` because `actions-rs` is deprecated
2. Added `FolderReader` for iterate over `FileReader`
3. Added `FileReader` for reader file from `cab`

## Breaking changes
1. Removed `Cabinet::get_file_entry`
2. Removed `Cabinet::read_file`

If the idea is ok I can finish it.

This is continue of: https://github.com/mdsteele/rust-cab/pull/20

### CODE REVIEW

File: src/file.rs [hunk 1]
```diff
@@ -59,87 +74,67 @@ impl FileEntry {
...
-        let bytes_read = self.reader.read(&mut buf[0..max_bytes])?;
+
+        let reader = &mut &self.reader.inner;
+        let bytes_read = reader.read(&mut buf[0..max_bytes])?;
         self.offset += bytes_read as u64;
         Ok(bytes_read)
     }
 }
 
-impl<'a, R: Read + Seek> Seek for FileReader<'a, R> {
```
Comment (mdsteele) [inline src/file.rs:117]: Is it still possible to support `Seek` for `FileReader`?

Comment (ikrivosheev) [reply src/file.rs:117]: @mdsteele, yes, it's possible. But it's bad for performance... PR doesn't finish, I want to add support `Cabinet::read_file`, I think I can do it.

File: examples/cabtool.rs [hunk 2]
```diff
@@ -42,21 +44,26 @@ fn main() {
     let cli = Cli::parse();
     match cli.command {
         Command::Cat { path, files } => {
-            let mut cabinet = Cabinet::new(File::open(path).unwrap()).unwrap();
+            let cabinet = Cabinet::new(File::open(path).unwrap()).unwrap();
             for filename in files {
-                let mut file_reader = cabinet.read_file(&filename).unwrap();
-                io::copy(&mut file_reader, &mut io::stdout()).unwrap();
+                while let Some(folder) = cabinet.folder_entries().next() {
```
Comment (mdsteele) [inline examples/cabtool.rs:49]: Is this line correct?  It seems like each loop iteration is creating a new `folder_entries()` iterator and starting over from the beginning.

(There's a few other loops that look like this too.)

**Review by mdsteele — COMMENTED** [review]
Sorry, finally found time to look at this again.  Thanks for your patience!

Just to check: is this PR still in progress, or is it ready to go?  (I think you mentioned in an earlier comment that you wanted to add `Cabinet::read_file`.)

**Comment by ikrivosheev** [comment]
> Sorry, finally found time to look at this again. Thanks for your patience!
> 
> Just to check: is this PR still in progress, or is it ready to go? (I think you mentioned in an earlier comment that you wanted to add `Cabinet::read_file`.)

Hello! Yes, you are right) I am working on add `Cabinet::read_file`. I think, I can do this.

File: examples/cabtool.rs (hunk 2 above)
Comment (ikrivosheev) [reply examples/cabtool.rs:49]: Ups, my bad. Good catch, I will fix it.

**Pull request closed without merging**